client/js/drawModule.js

```diff
@@ -0,0 +1,1031 @@
+  }
+
+  //create a form for creating or updating a review
+  function createReviewForm(restaurantName, review) {
+    var reviewForm = document.createElement('FORM');
+    reviewForm.name = 'editReviewForm';
+    reviewForm.className = 'editReviewForm';
+
+    reviewForm.onsubmit = function() {
+      var ratingValue =
+        parseInt(document.forms.editReviewForm.ratingValue.value);
+
+      var reviewBody = document.forms.editReviewForm.reviewBody.value;
+      if (review) {
+        updateReviewAction(review.name, ratingValue, reviewBody);
+      }
+      else {
+        createNewReviewAction(restaurantName, ratingValue, reviewBody);
+      }
+
+      return false;
+    };
+
+    var reviewLabel = document.createElement('LABEL');
+    reviewLabel.textContent = 'Your review: ';
+    reviewForm.appendChild(reviewLabel);
+    reviewForm.appendChild(document.createElement('BR'));
+
+    var reviewBody = document.createElement('TEXTAREA');
+    reviewBody.name = 'reviewBody';
+    reviewForm.appendChild(reviewBody);
+    reviewForm.appendChild(document.createElement('BR'));
+
+    var ratingLabel = document.createElement('LABEL');
+    ratingLabel.textContent = 'Rating value: ';
+    reviewForm.appendChild(ratingLabel);
+
+    var ratingValueSelect = document.createElement('SELECT');
+    ratingValueSelect.name = 'ratingValue';
+
+    var option;
+    for (var i = 0; i <= maxRating; i++) {
+      option = document.createElement('OPTION');
+      option.value = i;
+      option.textContent = i + ' Star' + (1 != i ? 's' : '');
+      ratingValueSelect.appendChild(option);
+    }
+
+    reviewForm.appendChild(ratingValueSelect);
+
+    var submit = document.createElement('INPUT');
+    submit.type = 'submit';
+    submit.value = 'Create Review';
+    submit.name = 'submitReview';
+    reviewForm.appendChild(submit);
+
+
+    return reviewForm;
+  }
+
+  //inicialize a review form when update a review
+  //instead of create a new one
+  function inicializeReviewForm(review) {
+
+    var reviewForm =
+      document.forms.namedItem('editReviewForm');
+
+    markSelectedValue(reviewForm.children.ratingValue,
+      review.reviewRating.ratingValue);
+
+    reviewForm.children.reviewBody.textContent =
+      review.reviewBody;
+
+    reviewForm.children.submitReview.value = 'Update review';
+  }
+
+
+
+  function addCreateReservationLink(restaurantName) {
+    var userInfo = JSON.parse(localStorage.getItem('userInfo'));
+
+    if (! connectionsAPI.hasRole(userInfo,
+      connectionsAPI.rol.endUser)) {
+      return null;
+    }
+
+    var createReservationLink = document.createElement('A');
+    createReservationLink.textContent = 'Make a reservation';
+    createReservationLink.onclick = (function(restaurantName) {
+      return function() {
+      createAndShowReservationForm(restaurantName);
+      };
+    })(restaurantName);
+
+    return createReservationLink;
+  }
+
+
+  //make a form for creating a reservation and initializes it
```

Comment:
"//make a form for creating a reservation and initializes it" -> 
//make a form to create a reservation and initialize it